src/IssueViz/IssueVisualizationControl/ViewModels/IssueVisualizationViewModel.cs

```diff
@@ -83,6 +84,9 @@ internal sealed class IssueVisualizationViewModel : IIssueVisualizationViewModel
                 selectionService.SelectedLocation);
         }
 
+        public bool HasNonNavigableLocations => CurrentIssue != null && 
+                                                CurrentIssue.GetAllLocations().Any(x => !x.IsNavigable());
+
```

Comment:
Wouldn't it be better to calculate and store the new value when one of the underlying properties changes, rather than recalculating on each get? I see the other properties on this class are calculated, but they are just calling simple property accessors. This one is more complex and will be causing new object allocations.

I think WPF can call the getters multiple times when rendering/repainting/resizing i.e. when the property hasn't changed.